src/routes/contract.js

```diff
@@ -29,22 +36,29 @@ module.exports = async (ctx) => {
       response.status = registrationStatus['blacklisted'];
       response.errors = await getContractErrors(nodeDb, contractId);
     } else {
-      const result = await getLastState(nodeDb, contractId);
+      const warpState = await getLastStateFromWarpCache(nodeDb, contractId);
+      let result = await getLastStateFromDreCache(nodeDb, contractId);
+      let parsed = false;
+      if (warpState && (!result || result.sort_key.localeCompare(warpState.sortKey) < 0)) {
+        warpState.cachedValue = JSON.parse(warpState.cachedValue);
+        result = await insertState(nodeDb, contractId, warpState);
```

Comment:
one small (or not?) issue I now see here is that `insertState` grabs the node's manifest. This manifest may change over time (e.g. when library versions change), so if this function will be called 'much' later than the state evaluation actually took place - there's a risk of signing the state with a different (from the one, that was actually used during evaluation) manifest.

But I guess we have to live with this for now.